File: nysol_python/mcmd/methods/convTex/db/smp/func_or.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='or'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,b1,b2,b3
1,1,0,0
2,1,,1
3,0,,0
4,0,0,0
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
'''
script['sh_code']='''
mcal c='or($b{b1},$b{b2},$b{b3})' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='or($b{b1},$b{b2},$b{b3})', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='ワイルドカードを利用した例'
script['comment']='''
``b`` から始まる項目( ``b1,b2,b3`` )をワイルドカード「 ``b*`` 」によって指定している。
'''
script['sh_code']='''
mcal c='or($b{b*})' a=rsl i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcal(c='or($b{b*})', a='rsl', i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_or.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,b1,b2,b3
1,1,0,0
2,1,,1
3,0,,0
4,0,0,0
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
EOF
scp=<<'EOF'
more dat1.csv
mcal c='or($b{b1},$b{b2},$b{b3})' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Example using wildcard "
comment=<<'EOF'
Use the wildcard character “\verb|b*|” to specify columns starting from \verb|b| (\verb|b1,b2,b3|).
EOF
scp=<<'EOF'
mcal c='or($b{b*})' a=rsl i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

